### PR TITLE
Add execute flag for __moservice allocations on zOS

### PR DESCRIPTION
**Pull request by joransiu**
A new `__MOPL_EXECUTABLE` flag is being defined to the zOS LE `__moservices()` call to specify executable property under PTF UI83032. Explicitly set the execute storage flags for memory allocated with __moservices if appropriate.

### CODE REVIEW

File: port/zos390/omrvmem.c [hunk 1]
```diff
@@ -87,6 +87,10 @@ int get_protectionBits(uintptr_t mode);
 
 #define GET_4K_ALIGNED_ALLOCATION_SIZE(byteAmount) (byteAmount + FOUR_K_MINUS_1 + sizeof(uintptr_t))
 
+#ifndef __MOPL_EXECUTABLE
+#define __MOPL_EXECUTABLE         0x04000000
+#endif
```
Comment (babsingh) [inline port/zos390/omrvmem.c:92]: Missing comment: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#comment-conditional-compilation-directives-if-ifdef-ifndef-else-elif-endif
```suggestion
#endif /* __MOPL_EXECUTABLE */
```

File: port/zos390/omrvmem.c [hunk 2]
```diff
@@ -845,6 +857,14 @@ reserve_memory_with_moservices(struct OMRPortLibrary *portLibrary, struct J9Port
 		Trc_PRT_vmem_reserve_memory_using_moservices_invalid_page_size(params->pageSize);
 		rc = -1;
 	}
+
+	/* Explicitly set the storage flag for executable. If left unspecified, the default is the storage will be executable,
+	 * unless DIAGxx IARNONEXEC64 trap is set. The __MOPL_EXECUTABLE flag is ignored on systems that do not support the flag.
```
Comment (babsingh) [inline port/zos390/omrvmem.c:862]: In the past, I was told not to exceed 100 chars in a single line for comments.

File: port/zos390/omrvmem.c [hunk 3]
```diff
@@ -845,6 +857,14 @@ reserve_memory_with_moservices(struct OMRPortLibrary *portLibrary, struct J9Port
...
+	 */
+	if (OMR_ARE_ANY_BITS_SET(OMRPORT_VMEM_MEMORY_MODE_EXECUTE, params->mode)) {
```
Comment (babsingh) [inline port/zos390/omrvmem.c:864]: The order of value and bits should be swapped as per the macro's definition:
https://github.com/eclipse/omr/blob/26e52210d25d5b3cd84b1573d4060ef7b663303e/include_core/omrcomp.h#L577
```suggestion
	if (OMR_ARE_ANY_BITS_SET(params->mode, OMRPORT_VMEM_MEMORY_MODE_EXECUTE)) {
```

**Review by babsingh — CHANGES_REQUESTED** [review]
The PR description has more details. The commit message does not match the PR description. Also, the commit message's body should be wrapped at 72 characters: https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#commit-guidelines.

**Comment by joransiu** [comment]
Thanks for the quick feedback @babsingh!  I should have addressed all your concerns in c7f7958.  Thanks!

**Comment by babsingh** [comment]
Thanks. Only the following feedback is pending: https://github.com/eclipse/omr/pull/6824#pullrequestreview-1189043489

> The PR description has more details. The commit message does not match the PR description. Also, the commit message's body should be wrapped at 72 characters: https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#commit-guidelines.

**Comment by joransiu** [comment]
> Thanks. Only the following feedback is pending: [#6824 (review)](https://github.com/eclipse/omr/pull/6824#pullrequestreview-1189043489)
> 
> > The PR description has more details. The commit message does not match the PR description. Also, the commit message's body should be wrapped at 72 characters: https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#commit-guidelines.

Sorry, somehow missed that comment.  I've updated the commit message text.

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
> https://ci.eclipse.org/omr/job/PullRequest-linux_ppc-64_le_gcc/3880/consoleFull

Known failure: https://github.com/eclipse/omr/issues/6571
```
16:07:26  30: [0;32m[----------] [m14 tests from Special/PPCDirectEncodingTest
16:07:26  30: free(): invalid next size (normal)
16:07:26  30/30 Test #30: compunittest ......................Child aborted***Exception:   0.89 sec
```

**Review by babsingh — APPROVED** [review]

**Pull request merged**